README.md
# Car Lease Calculator App

[![Car Lease & Finance Calculator App](/demo-image.png)](https://bit.ly/DealCalc)

A car lease & finance calculator web app for easily knowing whether you're getting a good deal on a lease.

## How It Works

See it in action: [bit.ly/DealCalc](https://bit.ly/DealCalc).

Simply enter your lease or finance numbers. The app calculates your monthly payment as well as other important information like total lease cost, total interest paid, etc. It also flags the deal's parameters that are less than ideal. The web app is built on top of [Lease-Calculator](https://github.com/ErezNagar/lease-calculator).

## CHANGELOG

See [CHANGELOG.md](https://github.com/ErezNagar/lease-calculator-app/blob/master/CHANGELOG.md)

## License

[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/).

## [1.4.1](https://github.com/ErezNagar/lease-calculator-app/compare/v1.3.1...v1.4.1) - 16/9/2023

### Update

- Using lease-calculator v4.1.0

## [1.3.1](https://github.com/ErezNagar/lease-calculator-app/compare/v1.3.0...v1.3.1) - 21/1/2023

### Fixed

- Sharing functionality on Safai on iOS

## [1.3.0](https://github.com/ErezNagar/lease-calculator-app/compare/v1.2.2...v1.3.0) - 2/10/2022

### Added

- Finance calculator. Calculates monthely payment, total finance cost, total principal and interest paid and due at signing amount
- Rating functionality to rate how this calcualtor is doing

## [1.2.2](https://github.com/ErezNagar/lease-calculator-app/compare/v1.2.1...v1.2.2) - 9/7/2021

### Fixed

- Fixed zer drive-off checkbox to work with query string input

## [1.2.1](https://github.com/ErezNagar/lease-calculator-app/compare/v1.2.0...v1.2.1) - 8/7/2021

### Fixed

- Fixed off MSRP percentage when Selling Price >= MSRP

## [1.2.0](https://github.com/ErezNagar/lease-calculator-app/compare/v1.1.0...v1.2.0) - 5/6/2021

### Added

- Now showing detailed list of drive off payments

## [1.1.0](https://github.com/ErezNagar/lease-calculator-app/compare/v1.0.2...v1.1.0) - 4/1/2021

### Added

- Support for zero drive-off (lease-calculator v2.0.0)

## [1.0.2](https://github.com/ErezNagar/lease-calculator-app/compare/v1.0.1...v1.0.2) - 23/12/2020

### Fixed

- Allow zero as a valid value
- Added a success state for share button

## [1.0.1](https://github.com/ErezNagar/lease-calculator-app/compare/v1.0.0...v1.0.1) - 6/12/2020

### Fixed

- Money Factor field to accept frictional values lower than 0.0005

## [1.0.0](https://github.com/ErezNagar/lease-calculator-app/compare/v0.5.1...v1.0.0) - 24/11/2020

### Changed

- New UI

### Added

- Support for multiple taxation methods
- Acquisition and disposition fees by manufacturer

## [0.5.1](https://github.com/ErezNagar/lease-calculator-app/compare/v0.5.1...v0.5.1) - 15/6/2020

### Fixed

- A bug in the sharing query string parsing where the pathname was missing

## [0.5.0](https://github.com/ErezNagar/lease-calculator-app/compare/v0.4.0...v0.5.0) - 15/6/2020

### Added

- Ability to share calculator results

## [0.4.0](https://github.com/ErezNagar/lease-calculator-app/compare/v0.3.1...v0.4.0) - 9/6/2020

### Added

- Ability to skip the wizard

## [0.3.1](https://github.com/ErezNagar/lease-calculator-app/compare/v0.3.0...v0.3.1) - 24/5/2020

### Fixed

- Down payment field now accepts zero as a valid value

## [0.3.0](https://github.com/ErezNagar/lease-calculator-app/compare/v0.2.3...v0.3.0) - 27/4/2020

### Changed

- Using official lease-calculator package release as dependency

## [0.2.3](https://github.com/ErezNagar/lease-calculator-app/compare/v0.2.2...v0.2.3) - 21/4/2020

### Fixed

- RV calculation

## [0.2.2](https://github.com/ErezNagar/lease-calculator-app/compare/v0.2.1...v0.2.2) - 21/4/2020

### Fixed

- RV calculation when switching between absolute and relative values

## [0.2.1](https://github.com/ErezNagar/lease-calculator-app/compare/v0.2.0...v0.2.1) - 16/4/2020

### Fixed

- Github repo link on results page (desktop)

## [0.2.0](https://github.com/ErezNagar/lease-calculator-app/compare/v0.1.0...v0.2.0) - 16/4/2020

### Added

- Tooltips for the price indicators on results page
- changelog.md

### Changed

- Moved all screens into the components folder
- Bigger navigation buttons on mWeb
- Changed the text on the results pages on mWeb

### Fixed

- Github repo link on results page

## [0.1.0](https://github.com/ErezNagar/lease-calculator-app/releases/tag/v0.1.0) - 14/4/2020

Initial version


src/constants.js
// Monthly payment out of MSRP, in percentage
export const MONTHLY_PAYMENT_TO_MSRP_THRESHOLD = 1;
// Discount off of MSRP, in percentage
export const OFF_MSRP_THRESHOLD = 10;
// Results onChange loading delay
export const RESULTS_CHANGE_DELAY = 10;

// Calculator type
export const CALCULATOR_TYPE_LEASE = "Lease";
export const CALCULATOR_TYPE_FINANCE = "Finance";

export const ESC_KEY = 27;
export const RIGHT_KEY = 39;

export const SHARE_BUTTON_TEXT = "Share your numbers";
export const SHARE_BUTTON_HELPER_TEXT = "Click to copy a shareable link";
export const SHARE_BUTTON_SUCCESS_CLICK_TEXT = "Link copied!";
export const SHARE_BUTTON_DELAY = 1000;
export const SHARE_BUTTON_SUCCESS_MESSAGE_DELAY = 3000;

export const MAKE = [
  {
    id: 1,
    displayName: "Acura",
  },
  { id: 3, displayName: "Audi" },
  {
    id: 4,
    displayName: "Bmw",
  },
  {
    id: 6,
    displayName: "Cadillac",
  },
  {
    id: 7,
    displayName: "Chevrolet",
  },
  {
    id: 8,
    displayName: "Chrysler",
  },
  {
    id: 9,
    displayName: "Dodge",
  },
  {
    id: 11,
    displayName: "Ford",
  },
  {
    id: 13,
    displayName: "GMC",
  },
  {
    id: 14,
    displayName: "Honda",
  },
  {
    id: 15,
    displayName: "Hyundai",
  },
  {
    id: 16,
    displayName: "Infiniti",
  },
  {
    id: 18,
    displayName: "Jeep",
  },
  {
    id: 19,
    displayName: "Kia",
  },
  {
    id: 20,
    displayName: "Land Rover",
  },
  {
    id: 21,
    displayName: "Lexus",
  },
  {
    id: 24,
    displayName: "Mazda",
  },
  {
    id: 25,
    displayName: "Mercedes benz",
  },
  {
    id: 26,
    displayName: "Mitsubishi",
  },
  {
    id: 27,
    displayName: "Nissan",
  },
  {
    id: 28,
    displayName: "Ram",
  },
  {
    id: 31,
    displayName: "Subaru",
  },
  {
    id: 33,
    displayName: "Toyota",
  },
  {
    id: 34,
    displayName: "Volkswagen",
  },
  {
    id: 35,
    displayName: "Volvo",
  },
];

// Tax applied on monthly payment
export const TAXATION_METHOD_MONTHLY = 1;
// Tax applied on sales price
export const TAXATION_METHOD_SALE_PRICE = 2;
// Taxed applied on total lease price
export const TAXATION_METHOD_TOTAL = 3;

export const DUMMY_DEFAULT_LEASE_DATA = {
  make: 33,
  msrp: 23000,
  sellingPrice: 21000,
  rv: 57,
  isRVPercent: true,
  mf: 0.00125,
  leaseTerm: 36,
  salesTax: 10.25,
  // totalFees: 1200,
  dealerFees: 900,
  governmentFees: 300,
  incentives: 500,
  // rebates: 500,
  isZeroDriveoff: false,
  taxMethod: TAXATION_METHOD_MONTHLY,
};

export const DUMMY_DEFAULT_FINANCE_DATA = {
  msrp: 50000,
  sellingPrice: 49000,
  salesTax: 6.25,
  financeTerm: 60,
  taxableFees: 250,
  untaxableFees: 300,
  APR: 4.5,
  downPayment: 0,
  tradeIn: 0,
  rebates: 0,
};
